packages/cloud-functions/src/cico/simplexEventPolling.ts

```diff
@@ -0,0 +1,99 @@
+import * as functions from 'firebase-functions'
+import { trackEvent } from '../bigQuery'
+import { SIMPLEX_DATA } from '../config'
+import { fetchWithTimeout, flattenObject, lookupAddressFromTxId } from './utils'
+
+const SIMPLEX_BIG_QUERY_EVENT_TABLE = 'cico_provider_events_simplex'
+
+// https://integrations.simplex.com/wallet-api-integration#events-api_group
+interface SimplexEventPayload {
+  events: SimplexTransactionEvent[]
+}
+
+interface SimplexTransactionEvent {
+  event_id: string
+  name: SimplexTxStatus
+  payment: {
+    id: string
+    status: string
+    created_at: string
+    partner_id: number
+    updated_at: string
+    crypto_currency: string
+    fiat_total_amount: {
+      amount: number
+      currency: string
+    }
+    crypto_total_amount: {
+      amount: number
+      currency: string
+    }
+    partner_end_user_id: string
+  }
+  timestamp: string
+}
+
+enum SimplexTxStatus {
+  Started = 'payment_request_submitted',
+  Completed = 'payment_simplexcc_approved',
+  Failed = 'payment_simplexcc_declined',
+}
+
+const getSimplexEvents = async () => {
+  const response = await fetchWithTimeout(SIMPLEX_DATA.event_url, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `ApiKey ${SIMPLEX_DATA.api_key}`,
+    },
+  })
+
+  if (!response.ok) {
+    throw new Error(JSON.stringify(response))
+  }
+
+  const simplexEvents: SimplexEventPayload = await response.json()
+  return simplexEvents
+}
+
+const deleteSimplexEvent = async (event: SimplexTransactionEvent) => {
+  try {
+    await fetchWithTimeout(`${SIMPLEX_DATA.event_url}/${event.event_id}`, {
+      method: 'DELETE',
+      headers: {
+        Authorization: `ApiKey ${SIMPLEX_DATA.api_key}`,
+      },
+    })
+  } catch (error) {
+    console.error('Delete error: ', JSON.stringify(error))
+    throw error
+  }
+}
+
+export const simplexEventPolling = functions.https.onRequest(async (_, res) => {
+  try {
+    const simplexEvents = await getSimplexEvents()
+
+    await Promise.all(
```

Comment:
How many `simplexEvents` do we expect to get each time this is called?

Reply:
The endpoint is queried every hour so it depends on how many users successfully attempted a Simplex cash-in in that time. The number has been fairly low (like < 10) but I can see it getting up to 20-30 if it gets more use